theano/tensor/signal/downsample.py

```diff
@@ -635,44 +634,70 @@ def c_code(self, node, name, inp, out, sub):
           Py_XDECREF(%(gx)s);
           %(gx)s = (PyArrayObject*) PyArray_ZEROS(4, PyArray_DIMS(%(x)s), x_typenum,0);
         }
```

Comment:
Do you mean we need to set gx to 0 even when the if in line 626 is not satisfied?